### DISCUSSION TITLE
Authentication Not Working

### DISCUSSION THREAD

**Discussion by karnie6** [rainbow-me/rainbowkit#856] [Q&A]
Trying to get authentication working on RainbowKit, but when I try signing the message, I see "error verifying signature", and it sends me to http://localhost:3000/api/auth/error (I made sure the connected wallet is the same one as signing)

Do i need to set up nextJS differently? (have a react/TS app).  Is this something common for localhost?

**Comment by QuintonC** [comment]
Can you provide some more information on your authentication stack and the file structure? What does your `pages/api/auth/[...nextauth].ts` look like?

**Reply by karnie6** [reply]
hey thanks @QuintonC .

Actually, i fixed this by copying the contents of [..nextauth].ts from the example on https://docs.login.xyz/integrations/nextauth.js, and it worked!

Thanks
Karthik
